### DISCUSSION TITLE
Remove orphans

### DISCUSSION THREAD

**Discussion by lukevandijk** [microsoft/vscode-docker#3945] [Q&A] [answered]
Type: <b>Feature Request</b>

When i change my docker compose to remove a service I no longer need I have ton manually delete this service in Docker Dashboard. 

Could you add the --remove-orphans flag to the execution. I believe this would classify as expected behaviour, when having removed a service from the compose spec.

Extension version: 1.25.0
VS Code version: Code 1.77.3 (Universal) (704ed70d4fd1c6bd6342c436f1ede30d1cff4710, 2023-04-12T09:19:37.325Z)
OS version: Darwin x64 22.4.0
Modes:
Sandboxed: Yes

**Comment by alexyaang** [comment, marked as answer]
I believe this could be done by [command customization](https://code.visualstudio.com/docs/containers/docker-compose#_command-customization). You can read more about it [here](https://code.visualstudio.com/docs/containers/reference#_command-customization) too. 

Here's an example for using command customization to add the `--remove-orphans` flag:

In `settings.json`:
```json
 "docker.commands.composeUp": [
        {
            "label": "override",
            "template": "${composeCommand} ${configurationFile} up ${detached} ${build} --remove-orphans",
        }
    ]
```

Let me know if this doesn't resolve your issue, then we can work something out :grin:
